common/http/src/retry.rs

```diff
@@ -245,44 +260,42 @@ impl RetrySender {
         let mut file_name = self.directory.clone();
         file_name.push(format!("{}_{}.retry.partial", fn_ts, fn_uuid));
 
-        let mut file = BufWriter::new(
-            OpenOptions::new()
-                .create(true)
-                .write(true)
-                .open(&file_name)
-                .await?,
-        );
-
-        // Manually serialize the body and offsets
-        file.write_all(b"{").await?;
-        let mut file = if let Some(offsets) = offsets {
-            file.write_all(b"\"offsets\":").await?;
-
-            // Serde can't write to async_write, so offload this to a threadpool
-            let std_file = tokio::task::spawn_blocking(
-                {
-                    // Get the std::fs::File out of the tokio BufWriter
-                    file.flush().await?;
-                    let mut std_file = std::io::BufWriter::new(file.into_inner().into_std().await);
-                    move || -> Result<std::fs::File, std::io::Error> {
-                        // Serialise the offsets to the std::io::BufWriter
-                        serde_json::to_writer(&mut std_file, &offsets)?;
-                        std_file.flush()?;
-                        Ok(std_file.into_inner()?)
+        {
+            let mut plain_file = BufWriter::new(
```

Comment:
Most idomatic rust code would keep the name as `file` so it's shaddowed on line 272 since you never refer to `plain_file` anywhere else in the method. Coming from Java/Scala/etc is does seem strange but does have the benefit that it's very clear that you do not intend to ever use the plain file object anywhere else other than in the creation of the gzip file. 

No change required, just FYI.